src/main/protobuf/soc/moves.proto

```diff
@@ -0,0 +1,112 @@
+syntax = "proto2";
+
+package protos.soc;
+
+import "soc/inventory.proto";
+import "soc/board.proto";
+import "soc/state.proto";
+
+enum ActionType {
+    ROLL_DICE = 0;
+    END_TURN = 1;
+    INITIAL_PLACEMENT = 2;
+    DISCARD_RESOURCES = 3;
+    MOVE_ROBBER_AND_STEAL = 4;
+    BUY_DEVELOPMENT_CARD = 5;
+    BUILD_ROAD = 6;
+    BUILD_SETTLEMENT = 7;
+    BUILD_CITY = 8;
+    PORT_TRADE = 9;
+    PLAY_KNIGHT = 10;
+    PLAY_YEAR_OF_PLENTY = 11;
+    PLAY_MONOPOLY = 12;
+    PLAY_ROAD_BUILDER = 13;
+}
+
+message HiddenCard {
+    repeated int32 viewable_by = 1;
+    oneof card {
+        Resource resource_card = 2;
+        DevelopmentCard development_card = 3;
+    }
+}
+
+message ResourceTransaction {
+    enum TransactionType {
+        GAIN = 0;
+        LOSE = 1;}
+    required TransactionType type = 1;
+    repeated ResourceCount resources = 2;
+}
+
+
+message ActionResult {
+    optional int32 roll = 1;
+    optional HiddenCard hidden_card = 2;
+    map<int32, ResourceTransaction> resource_transactions = 3;
+
+}
+
+message ActionEvent {
+    required ActionType type = 1;
```

Comment:
Can we move these into an `message ActionSpecification` type? 

``
message ActionSpecification
  GameAction action
  oneof payload {
    ...
  }
```

Reply:
I do not think we need an `ActionSpecification` type. That would result in a `GemeEvent` type that is: 
```
message GameEvent {
   ActionSpecification specification = 1;, 
  ActionResult result= 2;  
}
```
where `ActionType` in specification is shared between specification and result. Seems to be to be unnecessary nesting and narrows the scope of the type.

Reply:
That `GameEvent` looks great to me! What's bad about putting the `ActionType` inside of the action specification? I you wanted you could bubble it up:
```
message GameEvent {
   GameAction action = 1
   ActionSpecification specification = 2;, 
  ActionResult result= 3;  
}
```

Same goes for `Player` it could go inside the `ActionSpecification` or a level up.

The reason I *strongly* feel that `ActionSpecification` should be its own type, is that otherwise it gets duplicated between the `GameEvent` message and the "Move" message / rpc.

Does that make sense?

There should be a `makeMove` rpc or `takeAction` rpc or something like that and it should take an `ActionSpecification` message, the same message that is nested inside the `GameEvent`!